src/main/java/knightminer/simplytea/core/config/Teapot.java

```diff
@@ -15,6 +16,9 @@ public Teapot(ForgeConfigSpec.Builder builder) {
     fill_from_cauldron = builder.comment("If true, the teapot can be filled with water from a cauldron")
                                 .translation("simplytea.config.teapot.fill_from_cauldron")
                                 .define("fill_from_cauldron", true);
+    fill_from_fluid_tank = builder.comment("If true, the teapot can be filled with water from a fluid tank")
+                                  .translation("simplytea.config.teapot.fill_from_fluid_tank")
+                                  .define("fill_from_fluid_tank", true);
```

Comment:
I'm not sure this needs to be a config. The cauldron stuff was a config option mostly as cauldrons are super messy, though tbh we can clean up that cauldron code using the new vanilla API a ton. Feel free to update the cauldron logic in this PR or make a separate PR if you are interested, see `CauldronInteractions`.

I think for this, I would like the config to specify how much to fill. Default it to 1000, but a modpack might decide 250 makes more sense for a pot of water. If it goes down to 0 we could disable it I guess, as consuming 0 fluid does not make sense.

Reply:
Removed this config.

I also added a teapot_capacity config option, and changed the CauldronInteraction for teapots to lower the level of the cauldron according to the teapot_capacity by rounding to the nearest 1/3 bucket.

When this is updated to NeoForge 1.21 the CauldronInteraction for teapots can probably be removed in favor of using the CauldronWrapper with the fluid handling API.